include/libhal-arm-mcu/stm32_generic/i2c.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 Khalil Estell
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <libhal/output_pin.hpp>
+#include <libhal/units.hpp>
+#include <span>
+
+#include <libhal/i2c.hpp>
+#include <libhal/initializers.hpp>
+#include <libhal/io_waiter.hpp>
+
+namespace hal::stm32_generic {
+class i2c
+{
+public:
+  i2c(void* p_i2c, hal::io_waiter& p_waiter);
+  i2c();
+  void transaction(hal::byte p_address,
+                   std::span<hal::byte const> p_data_out,
+                   std::span<hal::byte> p_data_in,
+                   hal::function_ref<hal::timeout_function> p_timeout);
+  void configure(hal::i2c::settings const& p_settings, hertz p_frequency);
+
+  void clear_error_flag(hal::output_pin* sda, hal::output_pin* scl);
```

Comment:
```suggestion
  void clear_error_flag(hal::output_pin* sda, hal::output_pin* scl);
```

We spoke and this shouldn't be here. So lets remove this. :)